--- origsrc/OpenCV-2.0.0/3rdparty/flann/CMakeLists.txt	2009-09-30 19:21:00.000000000 -0500
+++ src/OpenCV-2.0.0/3rdparty/flann/CMakeLists.txt	2010-01-07 19:39:50.280381900 -0600
@@ -43,7 +43,7 @@ if(MSVC)
     add_definitions(-DJAS_WIN_MSVC_BUILD)
 endif()
 
-if(UNIX AND CMAKE_COMPILER_IS_GNUCXX)
+if(UNIX AND NOT CYGWIN AND CMAKE_COMPILER_IS_GNUCXX)
      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 endif()
--- origsrc/OpenCV-2.0.0/CMakeLists.txt	2009-09-30 19:21:05.000000000 -0500
+++ src/OpenCV-2.0.0/CMakeLists.txt	2010-01-07 20:06:57.886475600 -0600
@@ -584,7 +584,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     endif()
 
     # We need pthread's
-    if(UNIX)
+    if(UNIX AND NOT CYGWIN)
         set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -pthread")
     endif()
 
--- origsrc/OpenCV-2.0.0/OpenCVPCHSupport.cmake	2009-09-30 19:20:56.000000000 -0500
+++ src/OpenCV-2.0.0/OpenCVPCHSupport.cmake	2010-01-07 19:38:42.354496700 -0600
@@ -48,7 +48,7 @@ MACRO(_PCH_GET_COMPILE_FLAGS _out_compil
   IF(CMAKE_COMPILER_IS_GNUCXX)
 
     GET_TARGET_PROPERTY(_targetType ${_PCH_current_target} TYPE)
-    IF(${_targetType} STREQUAL SHARED_LIBRARY AND NOT WIN32)
+    IF(${_targetType} STREQUAL SHARED_LIBRARY AND NOT WIN32 AND NOT CYGWIN)
     	LIST(APPEND ${_out_compile_flags} "${${_out_compile_flags}} -fPIC")
     ENDIF()
 
