--- origsrc/OpenCV-2.0.0/CMakeLists.txt	2010-02-07 15:56:56.594333300 -0600
+++ src/OpenCV-2.0.0/CMakeLists.txt	2010-02-07 16:12:56.662246100 -0600
@@ -517,6 +517,7 @@ add_definitions(-DHAVE_CONFIG_H)
 
 message(STATUS "Parsing 'cvconfig.h.cmake'")
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cvconfig.h.cmake" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h")
+install(FILES "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h" DESTINATION include/opencv)
 
 # ---------------------------------------------------------------------------
 # The C+//0 include & link directories:
