--- origsrc/OpenCV-2.0.0/include/opencv/ml.h	2009-09-30 19:20:57.000000000 -0500
+++ src/OpenCV-2.0.0/include/opencv/ml.h	2010-01-07 19:44:46.058299400 -0600
@@ -351,7 +351,7 @@ struct CV_EXPORTS CvSVMParams
     CvSVMParams();
     CvSVMParams( int _svm_type, int _kernel_type,
                  double _degree, double _gamma, double _coef0,
-                 double _C, double _nu, double _p,
+                 double _Cv, double _nu, double _p,
                  CvMat* _class_weights, CvTermCriteria _term_crit );
 
     int         svm_type;
--- origsrc/OpenCV-2.0.0/src/cv/cvcalibration.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvcalibration.cpp	2010-01-07 20:48:20.931497700 -0600
@@ -42,6 +42,13 @@
 
 #include "_cv.h"
 
+#undef _A
+#undef _B
+#undef _J
+#undef _L
+#undef _M
+#undef _U
+
 /*
     This is stright-forward port v3 of Matlab calibration engine by Jean-Yves Bouguet
     that is (in a large extent) based on the paper:
--- origsrc/OpenCV-2.0.0/src/cv/cvlkpyramid.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvlkpyramid.cpp	2010-01-07 20:51:19.115689200 -0600
@@ -42,6 +42,10 @@
 #include <float.h>
 #include <stdio.h>
 
+#undef _A
+#undef _B
+#undef _M
+
 namespace cv
 {
 
--- origsrc/OpenCV-2.0.0/src/cv/cvlsh.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvlsh.cpp	2010-01-07 20:49:45.574339000 -0600
@@ -66,6 +66,8 @@
 #include <algorithm>
 #include <limits>
 
+#undef _L
+
 template <class T>
 class memory_hash_ops : public CvLSHOperations {
   int d;
--- origsrc/OpenCV-2.0.0/src/cv/cvshapedescr.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvshapedescr.cpp	2010-01-07 20:52:47.557747800 -0600
@@ -40,6 +40,10 @@
 //M*/
 #include "_cv.h"
 
+#undef _C
+#undef _S
+#undef _T
+
 /* calculates length of a curve (e.g. contour perimeter) */
 CV_IMPL  double
 cvArcLength( const void *array, CvSlice slice, int is_closed )
--- origsrc/OpenCV-2.0.0/src/cv/cvstardetector.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvstardetector.cpp	2010-01-07 20:53:21.631696700 -0600
@@ -41,6 +41,10 @@
 
 #include "_cv.h"
 
+#undef _I
+#undef _S
+#undef _T
+
 static void
 icvComputeIntegralImages( const CvMat* _I, CvMat* _S, CvMat* _T, CvMat* _FT )
 {
--- origsrc/OpenCV-2.0.0/src/cv/cvsurf.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvsurf.cpp	2010-01-07 20:54:04.416143900 -0600
@@ -109,6 +109,8 @@ Modifications by Ian Mahon
 */
 #include "_cv.h"
 
+#undef _X
+
 CvSURFParams cvSURFParams(double threshold, int extended)
 {
     CvSURFParams params;
--- origsrc/OpenCV-2.0.0/src/cv/cvundistort.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cv/cvundistort.cpp	2010-01-07 20:54:47.928632600 -0600
@@ -42,6 +42,9 @@
 
 #include "_cv.h"
 
+#undef _P
+#undef _R
+
 namespace cv
 {
 
--- origsrc/OpenCV-2.0.0/src/cvaux/cvba.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cvaux/cvba.cpp	2010-01-07 22:08:23.434185100 -0600
@@ -41,6 +41,9 @@
 
 #include "_cvaux.h"
 
+#undef _A
+#undef _B
+
 namespace cv {  
 
 LevMarqSparse::LevMarqSparse()
--- origsrc/OpenCV-2.0.0/src/cvaux/cvcalibfilter.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cvaux/cvcalibfilter.cpp	2010-01-07 22:15:21.912120700 -0600
@@ -42,8 +42,6 @@
 #include "_cvaux.h"
 #include <stdio.h>
 
-#undef quad
-
 #if _MSC_VER >= 1200
 #pragma warning( disable: 4701 )
 #endif
--- origsrc/OpenCV-2.0.0/src/cvaux/cvepilines.cpp	2009-09-30 19:20:58.000000000 -0500
+++ src/OpenCV-2.0.0/src/cvaux/cvepilines.cpp	2010-01-07 22:15:15.508754400 -0600
@@ -48,8 +48,6 @@
 
 /* Valery Mosyagin */
 
-#undef quad
-
 #define EPS64D 1e-9
 
 int cvComputeEssentialMatrix(  CvMatr32f rotMatr,
--- origsrc/OpenCV-2.0.0/src/cxcore/cxdxt.cpp	2009-09-30 19:20:59.000000000 -0500
+++ src/OpenCV-2.0.0/src/cxcore/cxdxt.cpp	2010-01-07 19:58:43.569202300 -0600
@@ -41,6 +41,8 @@
 
 #include "_cxcore.h"
 
+#undef log2
+
 namespace cv
 {
 
--- origsrc/OpenCV-2.0.0/src/cxcore/cxlapack.cpp	2009-09-30 19:20:59.000000000 -0500
+++ src/OpenCV-2.0.0/src/cxcore/cxlapack.cpp	2010-01-07 20:02:10.509038600 -0600
@@ -54,6 +54,7 @@
 #undef abs
 #undef max
 #undef min
+#undef _S
 
 namespace cv
 {
--- origsrc/OpenCV-2.0.0/src/cxcore/cxmatmul.cpp	2009-09-30 19:20:59.000000000 -0500
+++ src/OpenCV-2.0.0/src/cxcore/cxmatmul.cpp	2010-01-07 20:03:25.442324500 -0600
@@ -42,6 +42,9 @@
 
 #include "_cxcore.h"
 
+#undef _B
+#undef _C
+
 namespace cv
 {
 
--- origsrc/OpenCV-2.0.0/tests/cv/src/afundam.cpp	2009-09-30 19:20:56.000000000 -0500
+++ src/OpenCV-2.0.0/tests/cv/src/afundam.cpp	2010-01-07 22:17:06.460100500 -0600
@@ -41,6 +41,13 @@
 
 #include "cvtest.h"
 
+#undef _A
+#undef _I
+#undef _R
+#undef _U
+#undef _V
+#undef _W
+
 int cvTsRodrigues( const CvMat* src, CvMat* dst, CvMat* jacobian )
 {
     int depth;
