--- origsrc/OpenCV-2.0.0/interfaces/python/CMakeLists.txt	2009-09-30 19:20:57.000000000 -0500
+++ src/OpenCV-2.0.0/interfaces/python/CMakeLists.txt	2010-01-07 22:29:43.655409600 -0600
@@ -21,14 +21,14 @@ add_custom_command(
    )
 
 set(the_target "cvpy")
-add_library(${the_target} SHARED ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs} ${CMAKE_CURRENT_BINARY_DIR}/generated0.i)
+add_library(${the_target} MODULE ${lib_srcs} ${lib_hdrs} ${lib_int_hdrs} ${CMAKE_CURRENT_BINARY_DIR}/generated0.i)
 target_link_libraries(${the_target} ${PYTHON_LIBRARIES} cxcore cv highgui)
 set_target_properties(${the_target} PROPERTIES PREFIX "")
 set_target_properties(${the_target} PROPERTIES OUTPUT_NAME "cv")
 if(WIN32)
    set(CVPY_SUFFIX ".pyd")
 else()
-   set(CVPY_SUFFIX ".so")
+   set(CVPY_SUFFIX ${CMAKE_SHARED_MODULE_SUFFIX})
 endif()
 
 set_target_properties(${the_target} PROPERTIES SUFFIX ${CVPY_SUFFIX})
@@ -42,5 +42,5 @@ if(WIN32)
             ARCHIVE DESTINATION "Python${PYTHON_VERSION_MAJOR_MINOR}/Lib/site-packages" COMPONENT main
             )
 else()
-    install(FILES ${LIBRARY_OUTPUT_PATH}/cv.so DESTINATION ${PYTHON_PACKAGES_PATH})
+    install(TARGETS ${the_target} DESTINATION ${PYTHON_PACKAGES_PATH})
 endif()
