--- origsrc/OpenCV-2.3.0/CMakeLists.txt	2011-07-03 23:21:50.000000000 -0500
+++ src/OpenCV-2.3.0/CMakeLists.txt	2011-08-10 19:10:24.668956800 -0500
@@ -1009,7 +1009,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     endif()
 
     # We need pthread's
-    if(UNIX)
+    if(UNIX AND NOT CYGWIN)
       if(NOT ANDROID)
         set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -pthread")
       endif()
--- origsrc/OpenCV-2.3.0/OpenCVPCHSupport.cmake	2011-07-03 23:21:59.000000000 -0500
+++ src/OpenCV-2.3.0/OpenCVPCHSupport.cmake	2011-08-10 19:08:07.820129500 -0500
@@ -43,7 +43,7 @@ MACRO(_PCH_GET_COMPILE_FLAGS _out_compil
   IF(CMAKE_COMPILER_IS_GNUCXX)
 
     GET_TARGET_PROPERTY(_targetType ${_PCH_current_target} TYPE)
-    IF(${_targetType} STREQUAL SHARED_LIBRARY AND NOT WIN32)
+    IF(${_targetType} STREQUAL SHARED_LIBRARY AND NOT WIN32 AND NOT CYGWIN)
     	LIST(APPEND ${_out_compile_flags} "${${_out_compile_flags}} -fPIC")
     ENDIF()
 
