--- origsrc/OpenCV-2.3.0/modules/legacy/include/opencv2/legacy/legacy.hpp	2011-07-03 23:21:58.000000000 -0500
+++ src/OpenCV-2.3.0/modules/legacy/include/opencv2/legacy/legacy.hpp	2011-08-10 21:37:59.839443600 -0500
@@ -47,6 +47,8 @@
 #include "opencv2/features2d/features2d.hpp"
 #include "opencv2/calib3d/calib3d.hpp"
 
+#undef quad
+
 #ifdef __cplusplus
 extern "C" {
 #endif
--- origsrc/OpenCV-2.3.0/modules/traincascade/imagestorage.h	2011-07-03 23:21:59.000000000 -0500
+++ src/OpenCV-2.3.0/modules/traincascade/imagestorage.h	2011-08-10 20:39:29.453661000 -0500
@@ -1,6 +1,7 @@
 #ifndef _OPENCV_IMAGESTORAGE_H_
 #define _OPENCV_IMAGESTORAGE_H_
 
+#include <cstdio>
 #include "highgui.h"
 
 using namespace cv;
