--- origsrc/opencv-2.4.10/CMakeLists.txt	2014-10-01 02:33:36.000000000 -0500
+++ src/opencv-2.4.10/CMakeLists.txt	2014-12-16 12:35:47.862610200 -0600
@@ -325,7 +325,7 @@ else()
   set(OPENCV_INCLUDE_INSTALL_PATH "include")
 
   math(EXPR SIZEOF_VOID_P_BITS "8 * ${CMAKE_SIZEOF_VOID_P}")
-  if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
+  if(1) # LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
     set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/opencv)
   else()
     set(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
--- origsrc/opencv-2.4.10/cmake/OpenCVGenConfig.cmake	2014-10-01 02:33:36.000000000 -0500
+++ src/opencv-2.4.10/cmake/OpenCVGenConfig.cmake	2014-12-16 12:35:47.872610200 -0600
@@ -33,13 +33,11 @@ endif()
 
 
 
-if(WIN32)
-  if(MINGW)
+  if(MINGW OR CYGWIN)
     set(OPENCV_LINK_LIBRARY_SUFFIX ".dll.a")
-  else()
+  elseif(WIN32)
     set(OPENCV_LINK_LIBRARY_SUFFIX ".lib")
   endif()
-endif()
 
 #build list of modules available for the OpenCV user
 set(OpenCV_LIB_COMPONENTS "")
--- origsrc/opencv-2.4.10/cmake/templates/OpenCVConfig.cmake.in	2014-10-01 02:33:36.000000000 -0500
+++ src/opencv-2.4.10/cmake/templates/OpenCVConfig.cmake.in	2014-12-16 12:36:10.222641100 -0600
@@ -76,11 +76,7 @@ set(OpenCV_USE_MANGLED_PATHS @OpenCV_USE
 get_filename_component(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_FILE}" PATH CACHE)
 
 if(NOT WIN32 OR ANDROID)
-  if(ANDROID)
     set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../..")
-  else()
-    set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../..")
-  endif()
   # Get the absolute path with no ../.. relative marks, to eliminate implicit linker warnings
   if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_LESS 2.8)
     get_filename_component(OpenCV_INSTALL_PATH "${OpenCV_INSTALL_PATH}" ABSOLUTE)
